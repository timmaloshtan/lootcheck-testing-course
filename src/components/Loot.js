import React, { Component } from 'react';
import { connect } from 'react-redux';

import { fetchBitcoin } from '../actions/bitcoin';

export class Loot extends Component {
  componentDidMount() {
    this.props.fetchBitcoin();
  }
  
  render() {
    return (
      <div className="loot">
        <h3>BTC balance:</h3>
      </div>
    )
  }
}

const mapStateToProps = (state) => ({
  ...state,
});

const mapDispatchToProps = {
  fetchBitcoin,
};

export default connect(mapStateToProps, mapDispatchToProps)(Loot);
