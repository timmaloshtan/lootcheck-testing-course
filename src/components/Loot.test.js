import React from 'react';
import { shallow, mount } from 'enzyme';

import { Loot } from './Loot';

describe('Loot component', () => {
  let loot;
  const mockFetchBitcoin = jest.fn();
  const props = {
    balance: 10,
    bitcoin: {},
    fetchBitcoin: mockFetchBitcoin,
  };
  loot = shallow(<Loot {...props}/>);

  it('should render correctly', () => {
    expect(loot).toMatchSnapshot();
  });

  describe('when mounting', () => {
    beforeEach(() => {
      mockFetchBitcoin.mockClear();
      loot = mount(<Loot {...props} />);
    });

    it('should dispatch fetchBitcoin() from props', () => {
      console.log(mockFetchBitcoin.mock.calls);
      expect(mockFetchBitcoin).toHaveBeenCalled();
    });
  });
});
