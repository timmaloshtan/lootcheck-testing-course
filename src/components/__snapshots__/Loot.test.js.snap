// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loot component should render correctly 1`] = `
<div
  className="loot"
>
  <h3>
    BTC balance:
  </h3>
</div>
`;
